internal/bminventory/inventory.go

```diff
@@ -816,6 +816,10 @@ func (b *bareMetalInventory) GenerateClusterISO(ctx context.Context, params inst
 			WithPayload(common.GenerateError(http.StatusInternalServerError, formatErr))
 	}
 
+	if err := b.objectHandler.Upload(ctx, []byte(ignitionConfig), fmt.Sprintf("%s/discovery.ign", cluster.ID)); err != nil {
+		log.WithError(err).Warnf("Upload discovery ignition failed for cluster %s", cluster.ID)
```

Comment:
Why not Errorf and return 500?  Installations will depend on this.

Reply:
Because only some installations will depend on this.
I guess it's a pretty small chance, but if this fails and uploading the full iso wouldn't have failed I don't want to have failed that job if they were not going to use the ignition separately.

I don't feel strongly about this choice though so I can change it to a 500 :man_shrugging:

Reply:
It's unlikely that this will fail and the full upload will succeed, and even if so, it will leave the system in a weird state.  I'd rather fail and have the user retry.